File: leetcode-algorithms/611. Valid Triangle Number/solution.cpp
#include <iostream>
#include <algorithm>
#include <vector>
using namespace std;



int triangleNumber(vector<int>& nums) {
	sort(nums.begin(), nums.end());

	int res = 0;

	for(int i=2;i<nums.size();i++){
		int l=0,r=i-1;
		while(l<r){
			if(nums[l]+nums[r]>nums[i]){
				res += r-l;
				r--;
			}else{
				l++;
			}

		}
	}
	return res;
}


int main()
{
	vector<int> nums{2,2,3,4};
	int res = triangleNumber(nums);
	cout<<res<<endl;
	return 0;
}

File: leetcode-algorithms/611. Valid Triangle Number/solution.py
def triangleNumber(nums):
	"""
	:type nums: List[int]
	:rtype: int
	"""

	nums.sort()

	print(nums)

	res = 0

	for i in range(2,len(nums)):
		l = 0
		r = i - 1
		while l<r:
			if nums[l]+nums[r]>nums[i]:
				res += r - l
				r -= 1
			else:
				l += 1
	return res




if __name__ == '__main__':
	nums = [2,2,3,4]
	res = triangleNumber(nums)
	print(res)
